packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -0,0 +1,271 @@
+
+    expect(portfolio.portfolioId).toEqual('port-djh8932wr');
+  }),
+
+  test('fails portfolio from arn without resource name in arn', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    expect(() => {
+      Portfolio.fromPortfolioArn(stack, 'MyPortfolio', 'arn:aws:catalog:region:account-id:portfolio');
+    }).toThrowError(/Missing required Portfolio ID from Portfolio ARN/);
+  }),
+
+  test('portfolio share', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const p = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    p.share(shareAccountId);
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+    });
+  }),
+
+  test('portfolio share with share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, {
+      shareTagOptions: true,
+      acceptLanguage: AcceptLanguage.EN,
+    });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: true,
+      AcceptLanguage: 'en',
+    });
+  }),
+
+  test('portfolio share without share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, { shareTagOptions: false });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: false,
+    });
+  }),
+
+  test('portfolio share without explicit share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId);
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+    });
+  }),
+
+  test('portfolio principal association with role type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const role = new Role(stack, 'TestRole', {
+      assumedBy: new AccountRootPrincipal(),
+    });
+
+    portfolio.giveAccessToRole(role);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
```

Comment:
No need for the empty objects:

```suggestion
    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation');
````

Reply:
adjusted to add fields